src/psyclone/psyGen.py

```diff
@@ -3615,51 +3608,48 @@ def gen_arg_setter_code(self, parent):
         raise NotImplementedError("gen_arg_setter_code must be implemented "
                                   "by sub-class.")
 
-    def incremented_arg(self, mapping={}):
+    def incremented_arg(self):
         ''' Returns the argument that has INC access. Raises a
         FieldNotFoundError if none is found.
 
-        :param mapping: dictionary of access types (here INC) associated \
-                        with arguments with their metadata strings as keys
-        :type mapping: dict
-        :returns: a Fortran argument name.
         :rtype: str
         :raises FieldNotFoundError: if none is found.
 
         '''
-        assert mapping != {}, "psyGen:Kern:incremented_arg: Error - a "\
-            "mapping must be provided"
         for arg in self.arguments.args:
-            if arg.access.lower() == mapping["inc"]:
+            if arg.access == AccessType.INC:
                 return arg
+
+        api_config = Config.get().api_conf()
+        rev_access_mapping = api_config.get_reverse_access_mapping()
+
         raise FieldNotFoundError("Kernel {0} does not have an argument with "
                                  "{1} access".
-                                 format(self.name, mapping["inc"]))
+                                 format(self.name,
+                                        rev_access_mapping[AccessType.INC]))
```

Comment:
I've definitely not understood things yet. What's the difference between using `rev_access_mapping` and `api_name`? I'd naively have thought that they gave you the same information?

Reply:
No, you are spot on ... just that code was written before I decided to add the convenience function api_(specific)_name. I've updated this to use api_name() (and will change it once we agree on the final name for api_name()).

Reply:
Great!